src/Utils/obiDialogParser.ts

```diff
@@ -0,0 +1,331 @@
+                    if (!step.prompt) {
+                        throw new Error("Expected activity to be set")
+                    }
+                    const scorerStep = await this.getScorerStepfromActivity(step.prompt)
+                    trainRound.scorerSteps.push(scorerStep)
+                }
+                else if (step.$type === "Microsoft.BeginDialog") {
+                    const subDialog =  this.composerDialog.dialogs.find(d => d.$id === step.dialog)
+                    if (!subDialog) {
+                        throw new Error(`Dialog name ${step.dialog} undefined`)
+                    }
+
+                    const childDialogs = await this.getTrainDialogsfromOBIDialog(subDialog)
+                    
+                    // Add children to train dialog list
+                    trainDialogs = [...trainDialogs, ...childDialogs]
+                }
+                else if (step.$type !== "Microsoft.EndTurn") {
+                    console.log(`Unhandled OBI Type: ${step.$type}`)
+                }
+            }
+            if (trainRound) {
+                if (trainDialogs.length === 0) {
+                    trainDialogs.push(this.makeEmptyTrainDialog())
+                }
+                for (const td of trainDialogs) {
+                    // If susequent round has no extractor step, just prepend scorer steps
+                    if (td.rounds.length > 0 && td.rounds[0].extractorStep.textVariations.length === 0) {
+                        td.rounds[0].scorerSteps = [...trainRound.scorerSteps, ...td.rounds[0].scorerSteps]
+                    }
+                    // Otherwise prepend round
+                    else {
+                        td.rounds = [trainRound, ...td.rounds]
+                    }
+                }
+            }
+        }
+        return trainDialogs
+    }
+    
+    private async getScorerStepfromActivity(prompt: string): Promise<CLM.TrainScorerStep> {
+    
+        const parsedActivity = prompt.substring(prompt.indexOf("[") + 1, prompt.lastIndexOf("]")).trim()
+        let response = this.composerDialog.lgMap.get(parsedActivity)
+        if (!response) {
+            // LARS temp to handle badly create dialog
+            response = { text: "Can't Parse LU", suggestions: []}
+           //throw new Error(`LU name ${prompt} undefined`)
+        }
+    
+        let scoredAction: CLM.ScoredAction | undefined
+        if (this.createActionThunkAsync) {
+            const action = await this.getActionFromLG(response, true)
+            if (action) {
+                scoredAction = {
+                    actionId: action.actionId,
+                    payload: action.payload,
+                    isTerminal: action.isTerminal,
+                    actionType: CLM.ActionTypes.TEXT,
+                    score: 1
+                }
+            }
+        }
+    
+        let scoreInput: CLM.ScoreInput = {
+            filledEntities: [],  //LARS handle filled entities from api calls
```

Comment:
Did you means to check the `//LARS` comments in ?

Reply:
Yeah, they are a way for me to find actionable TODOs for myself.  We have a bunch of aspirational TODOs sprinkled in the code that make it hard to find things